include/beast/websocket/impl/stream.ipp

```diff
@@ -170,18 +170,29 @@ template<class NextLayer>
 template<class... Args>
 stream<NextLayer>::
 stream(Args&&... args)
-    : next_layer_(std::forward<Args>(args)...)
-    , stream_(next_layer_)
+    : stream_(std::forward<Args>(args)...)
 {
-    static_assert(is_Stream<next_layer_type>::value,
-        "Stream requirements not met");
+}
+
+template<class NextLayer>
+void
+stream<NextLayer>::
+accept()
+{
+    static_assert(is_SyncStream<next_layer_type>::value,
+        "SyncStream requirements not met");
```

Comment:
Great question. They are redundant. The thinking is that if they go off, the user will get an error message that is very close in the source file to the signature of the function they are trying to call, instead of a function with a different signature.